src/cc/bpf_module.cc

```diff
@@ -396,16 +389,27 @@ int BPFModule::annotate() {
         if (!fns.leaf_snprintf)
           errs() << "Failed to compile snprintf for " << *leaf_type << "\n";
 
-        ptrs_map[&it->second] = fns;
+        fn_ptrs_map_[&it->second] = fns;
       }
     }
   }
 
   rw_engine_ = finalize_rw(move(m));
+
+  return 0;
+}
+
+int BPFModule::finalize_annotate() {
+  fn_ptrs_lock_.lock();
```

Comment:
good suggestion. I tried scoped_lock which is not available for c++11, but I forgot to try lock_guard.